external/fv3fit/tests/training/test_main.py

```diff
@@ -0,0 +1,224 @@
+import dataclasses
+from typing import Optional, Sequence
+import fv3fit
+import fv3fit.train
+from fv3fit._shared.io import register
+import yaml
+import pytest
+import os
+import numpy as np
+import cftime
+import xarray as xr
+import loaders
+import vcm
+from unittest import mock
+
+
+@pytest.fixture
+def mock_dataset():
+
+    n_x, n_y, n_z, n_tile, n_time = (8, 8, 10, 6, 9)
+    arr = np.zeros((n_time, n_tile, n_z, n_y, n_x))
+    arr_surface = np.zeros((n_time, n_tile, n_y, n_x))
+    dims = ["time", "tile", "z", "y", "x"]
+    dims_surface = ["time", "tile", "y", "x"]
+
+    data = xr.Dataset(
+        {
+            "specific_humidity": (dims, arr),
+            "air_temperature": (dims, arr),
+            "downward_shortwave": (dims_surface, arr_surface),
+            "net_shortwave": (dims_surface, arr_surface),
+            "downward_longwave": (dims_surface, arr_surface),
+            "dQ1": (dims, arr),
+            "dQ2": (dims, arr),
+            "dQu": (dims, arr),
+            "dQv": (dims, arr),
+        },
+        coords={
+            "time": [
+                cftime.DatetimeJulian(2016, 8, day) for day in range(1, 1 + n_time)
+            ]
+        },
+    )
+
+    return data
+
+
+@dataclasses.dataclass
+class MainArgs:
+    data_path: str
+    train_config: str
+    train_data_config: str
+    val_data_config: str
+    output_path: str
+    local_download_path: Optional[str] = None
+
+
+class MockHyperparameters:
+    param1: str = ""
+
+
+@pytest.fixture
+def mock_train_dense_model():
+    train_mock = mock.MagicMock(name="train_dense_model")
+    train_mock.return_value = mock.MagicMock(
+        name="train_dense_model_return", spec=fv3fit.Predictor
+    )
+    register("mock")(train_mock.return_value.__class__)
+    original_func = fv3fit.get_training_function("DenseModel")
+    try:
```

Comment:
No, I thought it would but it won't. if you have cleanup you want to do in a fixture you need to put it in a `finally:` block it turns out, or it won't be triggered when the test fails.